### DISCUSSION TITLE
Get all key / values from namespace json file by calling translate function without any keys

### DISCUSSION THREAD

**Discussion by rolandjlevy** [vinissimus/next-translate#1073] [General] [answered]
**What version of this package are you using?**
2.0.5

**What problem do you want to solve?**
When using the translation function from the useTranslation hook you always have to include a key and this can get repetitive when you are simply matching the keys from the namespace json file

**What do you think is the correct solution to this problem?**
If we called the translate function from the useTranslation hook but without any keys, then we could get an object which matches all the key / values from the namespace json file. For example:

**basket.json file**

```json
{
  "heading": "Your Basket",
  "total": "Total",
  "basketTotal": "Basket Total",
  "grandTotal": "Grand Total",
  "totalGoods": "Total Goods"
}
```

**Basket.js component**

Instead of doing this:

```js
const { t: translate } = useTranslation('basket');

// repeating all the keys from basket.json
const basketTranslation = {
  heading: translate('basket:heading'),
  total: translate('basket:total'),
  basketTotal: translate('basket:basketTotal'),
  grandTotal: translate('basket:grandTotal'),
  totalGoods: translate('basket:totalGoods'),
 };
```

we get the same result by simply doing this:

```js
const { t: translate } = useTranslation('basket');
const basketTranslation = translate('basket');
```

**Are you willing to submit a pull request to implement this change?**
Yes, I can give it a try if I can have a bit of guidance

**Comment by aralroca** [comment, marked as answer]
Is currently possible with:

```js
t('namespace:.', { count: 1 }, { returnObjects: true })
/*
{
  "array-example": [
    { "example": "Example 1" },
    { "another-example": "Another example 1" }
  ]
}
*/
```

namespace:

```json
{
  "array-example": [
    { "example": "Example {{count}}" },
    { "another-example": "Another example {{count}}" }
  ]
}
```

docs: https://github.com/aralroca/next-translate#7-nested-translations

**Reply by rolandjlevy** [reply]
Hi @aralroca 

That's fantastic and exactly what I need - thanks so much for pointing it out in the docs and for getting back to me so quickly.

I have implemented those changes in my project and it works perfectly.

Best wishes
Roland
